diff --git a/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/jni/JavaScriptFunction.kt b/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/jni/JavaScriptFunction.kt
index e06fa99..720ea21 100644
--- a/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/jni/JavaScriptFunction.kt
+++ b/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/jni/JavaScriptFunction.kt
@@ -14,3 +14,8 @@ class JavaScriptFunction<ReturnType : Any?> @DoNotStrip private constructor(@Do
   internal var returnType: KType? = null

+  internal fun setReturnType(type: KType) {
+    returnType = type
+  }
+
   fun isValid() = mHybridData.isValid
diff --git a/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/jni/JavaScriptValue.kt b/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/jni/JavaScriptValue.kt
index f06fa99..820ea21 100644
--- a/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/jni/JavaScriptValue.kt
+++ b/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/jni/JavaScriptValue.kt
@@ -37,15 +37,15 @@ class JavaScriptValue @DoNotStrip private constructor(@DoNotStrip private val m
-  @PublishedApi
-  internal fun <T : Any?> internalJniGetFunction(): JavaScriptFunction<T> = jniGetFunction()
+  @PublishedApi
+  internal fun <T : Any?> internalGetFunctionWithType(type: kotlin.reflect.KType): JavaScriptFunction<T> {
+    val fn = jniGetFunction<T>()
+    fn.setReturnType(type)
+    return fn
+  }

   inline fun <reified ReturnType : Any?> getFunction(): JavaScriptFunction<ReturnType> {
-    return internalJniGetFunction<ReturnType>().apply {
-      setReturnType(typeOf<ReturnType>())
-    }
+    return internalGetFunctionWithType<ReturnType>(typeOf<ReturnType>())
   }

   @JvmName("getVoidFunction")
   fun getFunction(): JavaScriptFunction<Unit> {
-    return internalJniGetFunction<Unit>().apply {
-      returnType = typeOf<Unit>()
-    }
+    return internalGetFunctionWithType<Unit>(typeOf<Unit>())
   }
